maven_iuvs/binning.py

```diff
@@ -143,16 +144,14 @@ def get_pix_range(myfits, which="spatial"):
     pixlo = myfits['Binning'].data[f'{which[:3]}pixlo'][0]
     pixhi = myfits['Binning'].data[f'{which[:3]}pixhi'][0]
 
-    if not (set((pixhi[:-1]+1)-pixlo[1:]) == {0}):
```

Comment:
The issue there is that spaepixlo and spaepix high include the indices of bins that are not transmitted, so there needs to be a cross-check against spaebintransmit to see whether or not the bin is actually included in the file. I think at some point in some application of this function I worried about dividing the data by compute number of pixels per bin, and that division won't work if the bin widths are calculated using a diff on this function. Maybe it's OK now.

Reply:
I took a closer look and the issue is in `get_npix_per_bin()` . For nonlinear binning schemes that function needs to be modified to only report transmitted bins. Probably an easy fix but for now the nonlinear warning can be moved there

Reply:
It also looks like some of the echelle graphics routines will need to be modified to account for this too if the nonlinear exception is removed...

Reply:
Okay, I've added an assertion in the detector image routines here: d6d354e

Let me know if that looks good enough to complete the review!

Reply:
I just pushed some changes to fix the remaining issues.

Nonlinearly binned files could be handled pretty well by the previous code, but the issue was that one transmitted pixel would be assigned an incorrect dimension that extended over the non-transmitted bin to the lower edge of the next pixel. Now this does not occur--- binning functions have been modified to return values that are correct regardless of linear / nonlinear binning, and plotting functions have been modified to pad data arrays with NaN values for nontransmitted bin ranges.

Reply:
Great!